bindings/python/reader.py

```diff
@@ -219,7 +222,18 @@ def _timestamp_at_pos(self, pos_ptr):
         return ev.timestamp
 
     def _events(self, begin_pos_ptr, end_pos_ptr):
-        ctf_it_ptr = nbt._bt_ctf_iter_create(self._tc, begin_pos_ptr, end_pos_ptr)
+        if self.intersect_mode:
+            has_intersection = nbt._bt_python_has_intersection(self._tc)
+            if not has_intersection:
+                raise ValueError("No intersection found between trace files.")
```

Comment:
I'm not sure about throwing an exception here. If there is no intersection, I'd be tempted to return nothing; it may not be an error.